cli/flox/src/commands/environment.rs

```diff
@@ -1653,13 +1661,16 @@ impl Pull {
         dot_flox_path: PathBuf,
         pointer: ManagedPointer,
         force: bool,
-    ) -> Result<()> {
-        let mut env = ManagedEnvironment::open(flox, pointer, dot_flox_path)
-            .context("Could not open environment")?;
-        env.pull(force)?; //.context("Could not pull environment")?;
-        env.build(flox)?; //.context("Could not build environment")?;
-
-        Ok(())
+    ) -> Result<PullResult, EnvironmentError2> {
+        let mut env = ManagedEnvironment::open(flox, pointer, dot_flox_path)?;
+        match env.pull(force)? {
+            state @ PullResult::UpToDate => Ok(state),
+            // only build if the environment was updated
+            state @ PullResult::Updated => {
+                env.build(flox)?;
+                Ok(state)
+            },
+        }
```

Comment:
nit/slight preference
```suggestion
        let state = env.pull(force)?;
        // only build if the environment was updated
        if let PullResult::Updated = state {
            env.build(flox)?;
        }
        Ok(state)
```

Reply:
Agree, applied